File: src/main/scala/net/javajudd/scala/lang/variables/Variables.scala
package net.javajudd.scala.lang.variables

object Variables extends App {

  // Immutable variable
  val x = 1
  //x = 2;  // Reassignment to val
  println("Immutable variable x: " + x)

  // Mutable variable
  var y = 1;
  y = 2;
  println("Muttable variable y: " + y)
}

File: src/main/java/net/javajudd/java/variables/Variables.java
package net.javajudd.java.variables;

public class Variables {
    public static void main(String[] args) {

        // Immutable variable
        final int x = 1;
        // x = 2; // Cannot assign a value to final variable 'x'
        System.out.println("Immutable variable: x " + x);

        // Mutable variable
        int y = 1;
        y = 2;
        System.out.println("Mutable variable: y " + y);

        // Mutable var
        // NOTE: Requires Java 10+
        // var z = 1;
        // z = 2;
        // System.out.println("Mutable variable: z " + z);
    }
}
